packages/javascript/bh-shared-ui/src/views/Explore/ContextMenu/AssetGroupMenuItemPrivilegeZonesEnabled.tsx

```diff
@@ -16,95 +16,153 @@
+                            Cancel
+                        </Button>
+                        <Button variant='primary' onClick={onAccept} disabled={disableAccept}>
+                            Ok
+                        </Button>
+                    </DialogActions>
+                </DialogContent>
+            </DialogPortal>
+        </Dialog>
+    );
+};
+
+export const AssetGroupMenuItem: FC<{
+    assetGroupType: 'tierZero' | 'owned';
     showConfirmationOnAdd?: boolean;
-    confirmationOnAddMessage?: string;
-}> = ({
-    assetGroupId,
-    assetGroupName,
-    disableAddNode,
-    isCurrentMember,
-    removeNodePath,
-    onAddNode = () => {},
-    onShowConfirmation = () => {},
-    onCancelConfirmation = () => {},
-    showConfirmationOnAdd = false,
-    confirmationOnAddMessage = '',
-}) => {
-    const handleAddNode = () => {
-        onAddNode(assetGroupId);
-    };
-
-    const handleOnCancel = () => {
-        onCancelConfirmation();
-    };
-
-    // selected node is not a member of the group
-    if (!isCurrentMember) {
-        if (showConfirmationOnAdd) {
-            return (
-                <>
-                    <MenuItem onClick={onShowConfirmation}>Add to {assetGroupName}</MenuItem>
-                    <ConfirmNodeChangesDialog
-                        onCancel={handleOnCancel}
-                        onAccept={handleAddNode}
-                        dialogContent={confirmationOnAddMessage}
-                        disableAccept={disableAddNode}
-                    />
-                </>
-            );
-        } else {
-            return <MenuItem onClick={handleAddNode}>Add to {assetGroupName}</MenuItem>;
-        }
-    } else {
+}> = ({ assetGroupType, showConfirmationOnAdd = false }) => {
+    const [dialogOpen, setDialogOpen] = useState(false);
+    const { addNotification } = useNotifications();
+    const getAssetGroupTagsQuery = useTagsQuery();
+    const { selectedItemQuery } = useExploreSelectedItem();
+    const { checkPermission } = usePermissions();
+
+    const tierZeroAssetGroup = getAssetGroupTagsQuery.data?.find(
+        (value) => value.position === HighestPrivilegePosition
+    );
+    const ownedAssetGroup = getAssetGroupTagsQuery.data?.find((value) => value.type === AssetGroupTagTypeOwned);
+
+    const assetGroup = assetGroupType === 'tierZero' ? tierZeroAssetGroup : ownedAssetGroup;
+
+    const closeDialog = () => setDialogOpen(false);
+    const openDialog = () => setDialogOpen(true);
+
+    const { mutate: createSelector, isLoading: isMutationLoading } = useMutation({
+        mutationFn: () => {
+            const assetGroupId = assetGroup?.id ?? '';
+            const node = selectedItemQuery.data as NodeResponse;
+            return apiClient.createAssetGroupTagSelector(assetGroupId, {
+                name: node.label ?? node.objectId,
+                auto_certify: AssetGroupTagSelectorAutoCertifySeedsOnly,
```

Comment:
Heads up: This was actually bugged for the owned case, you can see the fix here: https://github.com/SpecterOps/BloodHound/pull/2150 but that went into hotfix / staging and not main. Hopefully my suggestion above will make this easy but regardless of the direction you go, the support for not having `auto_certify` in the payload for the owned case is a requirement for this feature to function correctly